include/ddc/memory.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
```

Comment:
Don't you think the introduction of the notion of memory-space would benefit from its own branch?

Reply:
Why not, we can add a step